qemu/tests/cpu_device_hotpluggable.py

```diff
@@ -0,0 +1,127 @@
+import logging
+
+from provider import cpu_utils
+
+from aexpect import ShellCmdError
+from virttest import error_context
+from virttest import utils_misc
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test hotplug/hotunplug of vcpu device.
+
+    1) Boot up guest w/o vcpu device.
+    2) Hot plug/unplug vcpu devices and check successfully or not. (qemu side)
+    3) Check if the number of CPUs in guest changes accordingly. (guest side)
+    4) Do sub test after hot plug/unplug.
+    5) Recheck the number of CPUs in guest.
+    6) Check the CPU topology of guest.
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+
+    def sub_reboot():
+        error_context.context("Reboot guest after vcpu %s"
+                              % hotpluggable_test, logging.info)
+        vm.reboot(method=params["reboot_method"], timeout=login_timeout)
+
+    def sub_shutdown():
+        error_context.context("Shutdown guest after vcpu %s"
+                              % hotpluggable_test, logging.info)
+        shutdown_method = params["shutdown_method"]
+        if shutdown_method == "shell":
+            session.sendline(params["shutdown_command"])
+            error_context.context("waiting VM to go down (guest shell cmd)",
+                                  logging.info)
+        elif shutdown_method == "system_powerdown":
+            vm.monitor.system_powerdown()
+            error_context.context("waiting VM to go down (qemu monitor cmd)",
+                                  logging.info)
+        if not vm.wait_for_shutdown(360):
```

Comment:
When it comes to "system_powerdown", we may need send the cmd twice. Sometimes guest doesn't respond the first time, and fail to go down. Tried to send cmd twice,  test passed every 20 times.

Reply:
OK, will sleep for 30s after CPU changes, if it still fails, I think it should a bug.